src/NLog/Layouts/XmlElementBase.cs

```diff
@@ -546,17 +549,39 @@ private void AppendXmlObjectPropertyValues(string propName, ref ObjectReflection
                         if (beforeValueLength > MaxXmlLength)
                             break;
 
-                        if (!AppendXmlPropertyObjectValue(property.Name, property.Value, property.TypeCode, sb, orgLength, objectsInPath, depth))
+                        var propertyTypeCode = property.TypeCode;
+                        if (propertyTypeCode != TypeCode.Object)
+                        {
+                            string xmlValueString = XmlHelper.XmlConvertToString((IConvertible)property.Value, propertyTypeCode, true);
+                            AppendXmlPropertyStringValue(property.Name, xmlValueString, sb, orgLength, false, ignorePropertiesElementName);
+                        }
+                        else
                         {
-                            sb.Length = beforeValueLength;
+                            if (!AppendXmlPropertyObjectValue(property.Name, property.Value, sb, orgLength, objectsInPath, depth))
+                            {
+                                sb.Length = beforeValueLength;
+                            }
                         }
                     }
                     AppendClosingPropertyTag(propNameElement, sb, ignorePropertiesElementName);
                 }
             }
         }
 
-        private string AppendXmlPropertyValue(string propName, object propertyValue, TypeCode objTypeCode, StringBuilder sb, int orgLength, bool ignoreValue = false, bool ignorePropertiesElementName = false)
+        private string AppendXmlPropertyValue(string propName, object propertyValue, StringBuilder sb, int orgLength, bool ignoreValue = false, bool ignorePropertiesElementName = false)
+        {
+            if (!ignoreValue)
+            {
+                string xmlValueString = XmlHelper.XmlConvertToStringSafe(propertyValue);
+                return AppendXmlPropertyStringValue(propName, xmlValueString, sb, orgLength, ignoreValue, ignorePropertiesElementName);
+            }
+            else
+            {
+                return AppendXmlPropertyStringValue(propName, string.Empty, sb, orgLength, ignoreValue, ignorePropertiesElementName);
```

Comment:
please move the return outside the if/else (the line is almost 100% duplicated)

Reply:
Done